### PR TITLE
[CDA-215] Users without 'Create Content' permission is able to edit CDA files

### PR DESCRIPTION
**Pull request by Advent51**
- permission check was added
- internationalization support was added

### CODE REVIEW

**Comment by Advent51** [comment]
**ATTENTION!**
The https://github.com/pentaho/pentaho-platform/pull/3834 should be merged first!

**Comment by Advent51** [comment]
Well, I run checkstyle locally and it is ok.
@mchen-len-son @pamval @dcleao please review

File: pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java [hunk 1]
```diff
@@ -469,13 +474,32 @@ public String clearCache( @Context HttpServletResponse servletResponse,
...
+  public boolean canCreateContent() {
+    IAuthorizationPolicy authorizationPolicy = SecurityHelper.getInstance().getAuthorizationPolicy();
+    if ( authorizationPolicy == null ) {
+      authorizationPolicy = PentahoSystem.get( IAuthorizationPolicy.class );
+      if ( authorizationPolicy == null ) {
+        logger.warn( "Couldn't retrieve Authorization Policy" );
+        return getUserSession().isAdministrator();
+      }
+    }
+    return authorizationPolicy.isAllowed( RepositoryCreateAction.NAME );
```
Comment (pamval) [inline pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java]: Is this the only way to get to the roles/permissions of the user?

I seem to remember we got there some other way. @afrjorge can you take a look?

Comment (Advent51) [reply pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java]: @pamval I used this code https://github.com/webdetails/cde/blob/feeef2e2c52ab9339df8cb6840973bc7e3dbd83d/cde-pentaho5/src/pt/webdetails/cdf/dd/PentahoCdeEnvironment.java#L218-L228 as example.
So if we are doing it in this way in cde, should we change it in cda?

Comment (pamval) [reply pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java]: The code in CDE does not need a new method in the ISecurityHelper interface. This one does. My question is why do we need to introduce a new method to that interface?

Comment (Advent51) [reply pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java]: Well, there is another working solution which is not causing the ISecurityHelper interface modification: https://github.com/Advent51/cda/commit/135a0e2f4e6253d8c7936c695f812291d5e6f288
But we need to add two more methods in ICdaEnvironment interface.
@pamval please review, may be the new solution is more appropriate.

Comment (pamval) [reply pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java]: yeah, I like that one better

**Pull request closed without merging**